lib/rucio/core/did_meta_plugins/did_column_meta.py

```diff
@@ -241,128 +180,141 @@ def list_dids(self, scope, filters, did_type='collection', ignore_case=False, li
-            elif k == 'length.lt':
-                query = query.filter(models.DataIdentifier.length < v)
-            elif k == 'length.gte':
-                query = query.filter(models.DataIdentifier.length >= v)
-            elif k == 'length.lte':
-                query = query.filter(models.DataIdentifier.length <= v)
-            elif k == 'length':
-                query = query.filter(models.DataIdentifier.length == v)
-            else:
-                query = query.filter(getattr(models.DataIdentifier, k) == v)
-
-        if 'name' in filters:
-            if '*' in filters['name']:
-                query = query.\
-                    with_hint(models.DataIdentifier, "NO_INDEX(dids(SCOPE,NAME))", 'oracle')
+        if not ignore_dids:
+            ignore_dids = set()
+
+        # mapping for semantic <type> to a (set of) recognised DIDType(s).
+        type_to_did_type_mapping = {
+            'all': [DIDType.CONTAINER, DIDType.DATASET, DIDType.FILE],
+            'collection': [DIDType.CONTAINER, DIDType.DATASET],
+            'container': [DIDType.CONTAINER],
+            'dataset': [DIDType.DATASET],
+            'file': [DIDType.FILE]
+        }
+
+        # backwards compatability for filters as single {}.
+        if isinstance(filters, dict):
+            filters = [filters]
+
+        # for each or_group, make sure there is a mapped "did_type" filter.
+        # if type maps to many DIDTypes, the corresponding or_group will be copied the required number of times to satisfy all the logical possibilities.
+        filters_tmp = []
+        for or_group in filters:
+            if 'type' not in or_group:
+                or_group_type = did_type.lower()
             else:
-                query = query.\
-                    with_hint(models.DataIdentifier, "INDEX(DIDS DIDS_PK)", 'oracle')
+                or_group_type = or_group.pop('type').lower()
+            if or_group_type not in type_to_did_type_mapping.keys():
+                raise exception.UnsupportedOperation('{} is not a valid type. Valid types are {}'.format(or_group_type, type_to_did_type_mapping.keys()))
+
+            for mapped_did_type in type_to_did_type_mapping[or_group_type]:
+                or_group['did_type'] = mapped_did_type
+                filters_tmp.append(or_group.copy())
+        filters = filters_tmp
+
+        # instantiate fe and create sqla query
+        fe = FilterEngine(filters, model_class=models.DataIdentifier)
+        query = fe.create_sqla_query(
+            additional_model_attributes=[
+                models.DataIdentifier.scope,
+                models.DataIdentifier.name,
+                models.DataIdentifier.did_type,
+                models.DataIdentifier.bytes,
+                models.DataIdentifier.length
+            ], additional_filters=[
+                (models.DataIdentifier.scope, operator.eq, scope),
+                (models.DataIdentifier.suppressed, operator.ne, true())
+            ]
+        )
```

Comment:
just copying over the existing core `list_dids` function (which uses the filter engine).